nowcasting_dataset/data_sources/gsp/gsp_data_source.py

```diff
@@ -429,11 +430,15 @@ def _get_central_gsp_id(
             # TODO: Implement finding GSP closest to x_center_osgb,
             # y_center_osgb.  This will probably be quite slow, so always
             # try finding an exact match first (which is super-fast).
-            raise NotImplementedError(
-                "Not yet implemented the ability to find GSP *nearest*"
-                " (but not at the identical location to) x_center_osgb and"
-                " y_center_osgb."
+
+            mat = cdist(
+                self.metadata[["location_x", "location_y"]],
+                [[x_center_osgb, y_center_osgb]],
+                metric="euclidean",
             )
+            closest_gsp = np.argmin(mat[:, 0])
+            # This is the closest GSP ID then
+            gsp_ids = [self.metadata.iloc[closest_gsp].gsp_id]
```

Comment:
nice, would be good to put a test for this. Even if you make an issue for the test